drivers/pwm/pwm_sam0.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Google LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * PWM driver using the SAM0 Timer/Counter (TCC) in Normal PWM (NPWM) mode.
+ * Supports the SAMD21 and SAMD5x series.
+ */
+
+#define DT_DRV_COMPAT atmel_sam0_tcc_pwm
+
+#include <device.h>
+#include <errno.h>
+#include <drivers/pwm.h>
+#include <soc.h>
+
+/* Static configuration */
+struct pwm_sam0_config {
+	Tcc *regs;
+	uint8_t channels;
+	uint8_t counter_size;
+	uint16_t prescaler;
+	uint32_t freq;
```

Comment:
I am not sure if this file should be called `pwm_sam0.c`.

In Atmel SAM, PWM may be implemented using either TC or TCC, and this driver only supports TCC; moreover, it would not be feasible to support both TC and TCC in one driver either due to the vast amount of differences between the two peripherals.

In fact, for very simple single channel PWM outputs like this, TC would be better suited; TCC is really intended for more advanced PWM controls like H-bridge motor drive).

This file should probably be called `pwm_sam0_tcc.c`.

Reply:
SGTM, will do.

Reply:
Done